inc/form.class.php

```diff
@@ -770,25 +770,7 @@ public static function getFormList(int $rootCategory = 0, string $keywords = '',
       if (PluginFormcreatorEntityConfig::getUsedConfig('is_kb_separated', Session::getActiveEntity()) != PluginFormcreatorEntityconfig::CONFIG_KB_DISTINCT
          && Session::haveRight('knowbase', KnowbaseItem::READFAQ)
       ) {
-         // Find FAQ entries
-         $params = [
-            'faq'      => '1',
-            'contains' => $keywords
-         ];
-         $params['knowbaseitemcategories_id'] = 0;
-         if (count($selectedCategories) > 0) {
-            $iterator = $DB->request($table_cat, [
-               'WHERE' => [
-                  'id' => $selectedCategories
-               ]
-            ]);
-            $kbcategories = [];
-            foreach ($iterator as $kbcat) {
-               $kbcategories[] = $kbcat['knowbaseitemcategories_id'];
-            }
-            $params['knowbaseitemcategories_id'] = $kbcategories;
```

Comment:
repalved with a QuerySubQuery object, to increase code readability  / performance